pkg/controller/jobs/job/job_multikueue_adapter.go

```diff
@@ -57,8 +58,10 @@ func (b *multikueueAdapter) SyncJob(ctx context.Context, localClient client.Clie
 	// the remote job exists
 	if err == nil {
 		if features.Enabled(features.MultiKueueBatchJobWithManagedBy) {
-			localJob.Status = remoteJob.Status
-			return localClient.Status().Update(ctx, &localJob)
+			return clientutil.PatchStatus(ctx, localClient, &localJob, false, func() (bool, error) {
```

Comment:
Hm, I think ultimately we will want the MultiKueue Jobs to be transparent to the users, as much as possible. For regular Jobs a user can add a custom condition and it is preserved. If we want to have the same logic for MultiKueue Jobs we would probably need to use SMP (Strategic Merge Patch), and make it strict. Otherwise we may drop user's conditions. 

IIUC dropping user conditions would also happen prior to this PR, so I'm not concerned. However, maybe using strict is more aligned with the long-term goal of mimicking the regular Job API. WDYT @alculquicondor ?

Reply:
However it will not make a big difference, if a spec change is relevant the code path will just go to workload recreation, if the spec changes are not relevant we are not interested at this point since we are just overwriting the status.

Reply:
Seeing a failed update in the logs in case a user tries to edit the Job status (say adding condition) might be useful for some debugging. It makes life simpler not to need to think about the race-conditions of changing spec / status. Even if some assumptions are correct now this could change. Also, as mentioned above, we would need to use strict to reconcile the status.

I'm not concerned with performance of strict vs non-strict. Using strict will not be regression since we used Udate prior to the PR.

OTOH, I get your point that it is probably ok now, and I don't see a concrete problematic scenario, so I wouldn't insist for strict=false, but it would be good to justify why we do it.

Reply:
Changed to `strict=true`.